src/Sulu/Bundle/AudienceTargetingBundle/EventListener/AudienceTargetingCacheListener.php

```diff
@@ -37,80 +36,43 @@ class HttpCache extends AbstractHttpCache
 
     const USER_CONTEXT_URL_HEADER = 'X-Forwarded-URL';
 
-    /**
-     * @var bool
-     */
-    private $hasAudienceTargeting;
+    protected $hadValidTargetGroupCookie = false;
 
     /**
-     * @param HttpKernelInterface $kernel
-     * @param bool $hasAudienceTargeting
-     * @param string $cacheDir
+     * @param CacheEvent $cacheEvent
      */
-    public function __construct(HttpKernelInterface $kernel, $hasAudienceTargeting = false, $cacheDir = null)
+    public function preHandle(CacheEvent $cacheEvent)
     {
-        parent::__construct($kernel, $cacheDir);
+        if (self::TARGET_GROUP_URL === $cacheEvent->getRequest()->getRequestUri()) {
+            return;
+        }
 
-        $this->hasAudienceTargeting = $hasAudienceTargeting;
+        $this->hadValidTargetGroupCookie = $this->setTargetGroupHeader(
+            $cacheEvent->getRequest(),
+            $cacheEvent->getKernel()
+        );
     }
 
     /**
-     * {@inheritdoc}
+     * @param CacheEvent $cacheEvent
      */
-    public function handle(Request $request, $type = HttpKernelInterface::MASTER_REQUEST, $catch = true)
+    public function postHandle(CacheEvent $cacheEvent)
     {
-        $hadValidTargetGroupCookie = null;
-        if ($this->hasAudienceTargeting) {
-            $hadValidTargetGroupCookie = $this->setTargetGroupHeader($request);
+        if (self::TARGET_GROUP_URL === $cacheEvent->getRequest()->getRequestUri()) {
+            return;
         }
 
-        $response = parent::handle($request, $type, $catch);
+        $response = $cacheEvent->getResponse();
+        $request = $cacheEvent->getRequest();
 
-        if (!$this->kernel->isDebug()) {
-            $response->headers->remove(self::HEADER_REVERSE_PROXY_TTL);
-        }
-
-        if ($this->hasAudienceTargeting && !$hadValidTargetGroupCookie) {
+        if (!$this->hadValidTargetGroupCookie) {
             $this->setTargetGroupCookie($response, $request);
         }
 
-        if ($this->hasAudienceTargeting && in_array(static::TARGET_GROUP_HEADER, $response->getVary())) {
+        if (in_array(static::TARGET_GROUP_HEADER, $response->getVary())) {
             $response->setMaxAge(0);
             $response->setSharedMaxAge(0);
```

Comment:
This seems to be overridden again later. See the following screenshot:
![image](https://user-images.githubusercontent.com/405874/29107379-49d68e34-7cda-11e7-99e8-bb9fa7c945eb.png)

Although there is a vary header, the shared max age is still set to 240.